### PR TITLE
fix(notifier): manage resourceType and provide a unregisterAll method…

### PR DESCRIPTION
**Pull request by leleueri**
… to remove trigger for a given resource at once

fixes gravitee-io/issues#7274

### CODE REVIEW

File: gravitee-node-notifier/src/main/java/io/gravitee/node/notifier/NotifierServiceImpl.java [hunk 1]
```diff
@@ -75,19 +82,73 @@ public void register(
...
+    final String resourceKey = buildResourceKey(
+      definition.getResourceId(),
+      definition.getResourceType()
+    );
+    List<NotificationDefinition> allDefinitions = definitions.get(resourceKey);
+    if (Objects.isNull(allDefinitions)) {
+      allDefinitions = Collections.synchronizedList(new ArrayList<>());
+      definitions.put(resourceKey, allDefinitions);
+    }
+    allDefinitions.add(definition);
```
Comment (jhaeyaert) [inline gravitee-node-notifier/src/main/java/io/gravitee/node/notifier/NotifierServiceImpl.java:110]: Maybe I'm wrong but I suspect a possible side effect if 2 `NotificationDefinition` on the same resource are registered almost in the same time (the `allDefinitions` list could be erased and one of the NotificationDefinition can be lost).

Do you think it is possible to rework it using a `computeIfAbsent` instead and rely on the concurrent map to ensure nothing could be lost ?

**Review by jhaeyaert — CHANGES_REQUESTED** [review]
Only one thing to check regarding concurrency otherwise everything looks good to me.

File: gravitee-node-notifier/src/main/java/io/gravitee/node/notifier/NotifierServiceImpl.java [hunk 2]
```diff
@@ -75,19 +82,76 @@ public void register(
...
+    List<NotificationDefinition> allDefinitions = definitions.get(resourceKey);
+    if (Objects.isNull(allDefinitions)) {
+      allDefinitions =
+        definitions.computeIfAbsent(
+          resourceKey,
+          key -> Collections.synchronizedList(new ArrayList<>())
+        );
+    }
+    allDefinitions.add(definition);
+  }
```
Comment (jhaeyaert) [inline gravitee-node-notifier/src/main/java/io/gravitee/node/notifier/NotifierServiceImpl.java:114]: I think the problem remains the same no ?
What about this:

```java
definitions.computeIfAbsent(
          resourceKey,
          key -> Collections.synchronizedList(new ArrayList<>())
        ).add(definition);
```

Comment (leleueri) [reply gravitee-node-notifier/src/main/java/io/gravitee/node/notifier/NotifierServiceImpl.java:114]: :face_exhaling: :+1:

**Review by jhaeyaert — APPROVED** [review]

**Pull request merged**